sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -117,6 +120,9 @@ public SerializerAdapter serializer() {
      * @return a {@link Mono} that emits HttpResponse asynchronously
      */
     public Mono<HttpResponse> send(HttpRequest request, Context contextData) {
+        if(request.body() != null){
+            request.body().map(ByteBuffer::reset);
```

Comment:
Is this happening because the `UnexpectedLengthException` is reading the `ByteBuffer`? If so, is there any way the check could be implemented to not read the ByteBuffer and need it to be reset?

Reply:
Actually, it looks like we don't do any reading of the `ByteBuffer`, so can this be removed?

Reply:
It is not needed. Good catch. Will remove,